### ISSUE TITLE
NaN occurs when building with Dropout, Conv2DTranspose, LeakyReLU, ELU, PReLU, Flatten and Dense

### ISSUE THREAD

**Issue by bugreporter450** [tensorflow/tensorflow#46624]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:a GNU/Linux system with Linux kernel 4.15.0 on 1 6-core 3.60GHz Intel Core CPU i7-6850K with 64 GB RAM equipped with a NVIDIA Corporation GP102 GPUs
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below):tensorflow2-GPU
- Python version:3.6

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I got the NaN loss when I tried to train my model

**Describe the expected behavior**
The loss should be a number

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
```
batch_size = 56
epochs = 75
num_classes = 10
import os
save_dir = 'model'
model_name = 'trained_model.h5'
import tensorflow.keras as keras
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()
img_rows, img_cols = x_train.shape[1], x_train.shape[2]

x_train = x_train.reshape(x_train.shape[0], img_rows, img_cols, 1)
x_test = x_test.reshape(x_test.shape[0], img_rows, img_cols, 1)
input_shape = (img_rows, img_cols, 1)
x_train = x_train.astype('float32')
x_test = x_test.astype('float32')
x_train /= 255
x_test /= 255
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)
import time
begin_time = time.time()
import tensorflow.keras as keras
model = keras.models.Sequential()
model.add(keras.layers.Dropout(rate = 0.21019177254509236,noise_shape=None, seed = 5))
model.add(keras.layers.Conv2DTranspose(filters=7,kernel_size=(11, 11), strides=(11, 11), padding='same',activation='tanh',kernel_initializer='Constant'))
model.add(keras.layers.LeakyReLU(alpha=0.28649457154482805))
model.add(keras.layers.ELU(alpha=0.7631891438465761))
model.add(keras.layers.PReLU(alpha_initializer='Constant'))

model.add(keras.layers.Flatten())
model.add(keras.layers.Dense(num_classes, activation='selu'))
model.compile(loss=keras.losses.categorical_crossentropy,optimizer=keras.optimizers.Adadelta(), metrics=['accuracy'])
model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, verbose=1, validation_data=(x_test, y_test))
start_time = time.time()
score = model.evaluate(x_test, y_test, verbose=0)
end_time = time.time()
time_data = end_time - start_time
print('Training Time costs', start_time - begin_time)
print('Tesing Time costs', time_data)
print('Test accuracy:', score[1])
model_path = os.path.join(save_dir, model_name)
model.save(model_path)
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

```
Epoch 175
10721072 [==============================] - 641s 598msstep - loss nan - accuracy 0.0987 - val_loss nan - val_accuracy 0.0980
Epoch 275
10721072 [==============================] - 651s 607msstep - loss nan - accuracy 0.0987 - val_loss nan - val_accuracy 0.0980
Epoch 375
10721072 [==============================] - 671s 626msstep - loss nan - accuracy 0.0987 - val_loss nan - val_accuracy 0.0980
Epoch 475
10721072 [==============================] - 684s 638msstep - loss nan - accuracy 0.0987 - val_loss nan - val_accuracy 0.0980
Epoch 575
 7601072 [====================.........] - ETA 307 - loss nan - accuracy 0.0989
```

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with [TF v2.4](https://colab.research.google.com/gist/amahendrakar/e43abdc536cb6a95597f7cf99165be7f/46624.ipynb) and [TF-nightly](https://colab.research.google.com/gist/amahendrakar/95743a9e7c13c31a4c68a66fd227cf98/46624-tf-nightly.ipynb).

Whereas with [TF v2.3](https://colab.research.google.com/gist/amahendrakar/e67f96992550fefa438da0eb01e76c24/46624-2-3.ipynb#scrollTo=kWLP9rDeOL9Q), the loss value is constant at 1.1921e-07. Please check the linked gist for reference. Thanks!

**Comment by ymodak** [comment]
Changing `kernel_initializer` of `Conv2DTranspose` helps overcome the `NaN` values.
I have tried with `glorot_uniform` and `LecunNormal` initializers.
```python
model.add(keras.layers.Conv2DTranspose(filters=7,kernel_size=(11, 11), strides=(11, 11), padding='same',activation='tanh',kernel_initializer='glorot_uniform'))
```

**Comment by nikitamaia** [comment]
Consider changing the model's last layer to have a softmax activation (instead of selu) since this is a classification problem.
`model.add(keras.layers.Dense(num_classes, activation='selu'))`
